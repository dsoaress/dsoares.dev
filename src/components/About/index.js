import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'

import BlockContent from '../BlockContent'
import Social from '../Social'
import * as S from './styled'

const About = () => {
  const { sanitySiteSettings } = useStaticQuery(
    graphql`
      query {
        sanitySiteSettings {
          bio {
            title
            image {
              asset {
                fluid(maxWidth: 420) {
                  ...GatsbySanityImageFluid
                }
              }
            }
            _rawText
          }
        }
      }
    `
  )

  return (
    <S.Wrapper>
      <h1>{sanitySiteSettings.bio.title}</h1>
      <S.Grid>
        <S.TextWrapper>
          {sanitySiteSettings.bio._rawText && (
            <BlockContent blocks={sanitySiteSettings.bio._rawText} />
          )}
          <Social />
        </S.TextWrapper>
        <S.Image fluid={sanitySiteSettings.bio.image.asset.fluid} />
      </S.Grid>
    </S.Wrapper>
  )
}

export default About
