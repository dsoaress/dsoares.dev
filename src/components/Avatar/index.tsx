import Image from 'next/image'

import avatar from '@/assets/avatar.jpg'
import profileData from '@/content/profile.json'

import { Overlay, Wrapper } from './styles'

export function Avatar() {
  return (
    <Wrapper>
      <Image
        src={avatar}
        alt={profileData.title}
        objectFit="cover"
        objectPosition="top"
        layout="fill"
        placeholder="blur"
        quality={100}
      />
      <Overlay />
    </Wrapper>
  )
}
