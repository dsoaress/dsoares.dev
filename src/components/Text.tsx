import type { CSS, VariantProps } from '@stitches/react'
import type { HTMLAttributes } from 'react'

import { sharedTypographyProps, sharedTypographyVariants } from '@/styles/shared'
import { styled } from '@/styles/stitches'

type TextProps = {
  as?: 'p' | 'span'
  css?: CSS
} & HTMLAttributes<HTMLParagraphElement> &
  VariantProps<typeof StyledText>

const StyledText = styled('p', sharedTypographyProps, sharedTypographyVariants, {
  defaultVariants: {
    size: 'md'
  }
})

export function Text(props: TextProps) {
  return <StyledText {...props} />
}
