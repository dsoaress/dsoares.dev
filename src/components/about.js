import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'

import BlockContent from './block-content'
import Social from './social'

const About = () => {
  const { sanitySiteSettings } = useStaticQuery(
    graphql`
      query {
        sanitySiteSettings {
          bio {
            title
            image {
              asset {
                fluid(maxWidth: 420) {
                  ...GatsbySanityImageFluid
                }
              }
            }
            _rawText
          }
        }
      }
    `
  )

  return (
    <div>
      <h1>{sanitySiteSettings.bio.title}</h1>
      <div>
        <div>
          {sanitySiteSettings.bio._rawText && (
            <BlockContent blocks={sanitySiteSettings.bio._rawText} />
          )}
          <Social />
        </div>
        <Img fluid={sanitySiteSettings.bio.image?.asset.fluid} />
      </div>
    </div>
  )
}

export default About
