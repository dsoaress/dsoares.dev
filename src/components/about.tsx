import { Block } from './block'

export function About() {
  return (
    <Block className="col-span-12 text-3xl leading-snug">
      <p className="text-neutral-400">
        Specializing in <span className="text-neutral-50">scalable, robust applications</span>, I
        focus on delivering seamless user experiences and optimizing backend systems for maximum
        efficiency.
      </p>
    </Block>
  )
}
