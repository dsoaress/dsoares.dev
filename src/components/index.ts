export * from './about'
export * from './block'
export * from './footer'
export * from './header'
export * from './home-container'
export * from './logo'
export * from './social'
