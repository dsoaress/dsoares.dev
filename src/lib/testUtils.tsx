import { render } from '@testing-library/react'
import { ReactElement, ReactNode } from 'react'

type ProvidersProps = {
  children?: ReactNode
}

const Providers = ({ children }: ProvidersProps) => {
  return <>{children}</>
}

const customRender = (ui: ReactElement, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
