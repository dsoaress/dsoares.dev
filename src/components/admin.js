import { useState } from 'react'
import useSWR, { mutate } from 'swr'
import { signOut } from 'next-auth/client'

import Spinner from '@/components/spinner'
import { fetcher } from '@/lib/fetcher'

export default function Tasks() {
  const [label, setLabel] = useState('')
  const [url, setUrl] = useState('')

  const { data } = useSWR('/api', fetcher, {
    refreshInterval: 500
  })

  if (!data) return <Spinner />

  async function handleSubmit(e) {
    e.preventDefault()
    mutate('/api', { ...data, data: { links: { label, url } } }, false)

    await fetch('/api', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ link: { label, url } })
    })

    mutate('/api')
    setLabel('')
    setUrl('')
  }

  return (
    <div>
      <div>
        <button onClick={() => signOut()}>Sign out</button>
      </div>
      <form onSubmit={handleSubmit}>
        <div className="text-black">
          <label>Label</label>
          <input
            type="text"
            value={label}
            onChange={e => setLabel(e.target.value)}
            required
          />
          <label>Url</label>
          <input
            type="text"
            value={url}
            onChange={e => setUrl(e.target.value)}
            required
          />
        </div>
        <button type="submit">Add</button>
      </form>
      <div>
        {!data.links || data.links.length === 0 ? (
          <p>
            It looks like you don't have any more tasks... Would you like to add
            a new one?
          </p>
        ) : (
          data.links.map(link => (
            <div key={link._id}>
              <p>{link.label}</p>
              <p>{link.url}</p>
            </div>
          ))
        )}
      </div>
    </div>
  )
}
