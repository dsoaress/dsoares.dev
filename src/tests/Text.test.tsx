import { Text } from '@/components/Text'
import { render, screen } from '@/lib/testUtils'

describe('<Text />', () => {
  it('Renders with children', () => {
    render(<Text>Children</Text>)

    expect(screen.getByText('Children')).toBeInTheDocument()
    expect(screen.getByText('Children')).toMatchSnapshot()
  })
})
