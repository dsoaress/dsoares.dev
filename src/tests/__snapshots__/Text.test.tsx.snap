// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> Renders with children 1`] = `
.c0 {
  line-height: 1.75rem;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 300;
}

<p
  class="c0"
>
  Children
</p>
`;
